engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2405,6 +2540,46 @@ private LogicalOperatorInfo fillLogicalOperatorInfo(OperatorMeta operator)
     return loi;
   }
 
+  private LogicalModuleInfo fillLogicalModuleInfo(ModuleMeta module, boolean flatten, String parentModuleName)
+  {
+    LogicalModuleInfo lmi = new LogicalModuleInfo();
+    lmi.name = module.getName();
+    lmi.className = module.getModule().getClass().getName();
+    if (flatten) {
+      for (OperatorMeta operatorMeta : getLogicalPlan().getAllOperators()) {
```

Comment:
Yes you are correct.

In the top level dag,
operator O1 in module M1 has the name "M1:O1" with module name "M1"
operator O2 with name "M1:M2:O2" and module name "M1:M2" 

In dag of M1
operator O1 with name "O1" with module name as null
operator O2 with name "M2:O2" with module name "M2"

When we are iterating dag M1 we need to find out the operatorMeta in the top level DAG. I have refactored this code to remove iteration on top level DAG.